src/expr/src/relation/func.rs

```diff
@@ -2001,6 +2002,105 @@ fn unnest_list<'a>(a: Datum<'a>) -> impl Iterator<Item = (Row, Diff)> + 'a {
         .map(move |e| (Row::pack_slice(&[e]), 1))
 }
 
+// transliteration from postgres/src/backend/utils/adt/misc.c
+fn parse_ident<'a>(
```

Comment:
One potential concern is that this function may diverge from our logic for parsing identifiers in the parser. However, I think that's fine because it looks like PostgreSQL also has different code paths for this function and parsing identifiers in their parser.

I just wanted to write that down in case someone looks through this later.

Reply:
The lexer module is private, which makes this very challenging to share.